include/deal.II/multigrid/mg_transfer_global_coarsening.h

```diff
@@ -0,0 +1,444 @@
+     * Polynomial degree of the finite element of the coarse cells.
+     */
+    unsigned int degree_coarse;
+
+    /**
+     * Polynomial degree of the finite element of the fine cells.
+     */
+    unsigned int degree_fine;
+
+    /**
+     * Flag if the finite element on the fine cells are contiguous. If yes,
+     * the multiplicity of DoF shares and constraints have to be taken in
+     * account via weights.
+     */
+    bool fine_element_is_continuous;
+
+    /**
+     * Weights for continuous elements.
+     */
+    std::vector<Number> weights;
+
+    /**
+     * 1D prolongation matrix.
+     */
+    AlignedVector<VectorizedArray<Number>> prolongation_matrix_1d;
+
+    /**
+     * DoF indices of the coarse cells.
+     */
+    std::vector<unsigned int> level_dof_indices_coarse;
+
+    /**
+     * DoF indices of the fine cells.
+     */
+    std::vector<unsigned int> level_dof_indices_fine;
+
+    /**
+     * Print internal data structures to stream @p out.
+     */
+    template <typename Stream>
+    void
+    print(Stream &out) const;
+  };
+
+  /**
+   * Transfer schemes.
+   */
+  std::vector<MGTransferScheme> schemes;
+
+  /**
+   * Partitioner needed by the intermediate vector.
+   */
+  std::shared_ptr<const Utilities::MPI::Partitioner> partitioner_fine;
+
+  /**
+   * Partitioner needed by the intermediate vector.
+   */
+  std::shared_ptr<const Utilities::MPI::Partitioner> partitioner_coarse;
+
+  /**
+   * Internal vector needed for collecting all degrees of freedom of the
+   * fine cells.
+   */
+  mutable LinearAlgebra::distributed::Vector<Number> vec_fine;
+
+  /**
+   * Internal vector on that the actual prolongation/restriction is performed.
+   */
+  mutable LinearAlgebra::distributed::Vector<Number> vec_coarse;
+
+  /**
+   * Constraint matrix on coarse level.
+   */
+  AffineConstraints<Number> constraint_coarse;
+
+  friend class internal::MGTwoLevelTransferImplementation;
+};
+
+
+
+/**
+ * Implementation of the MGTransferBase. In contrast to
+ * other multigrid transfer operators, the user can provide separate
+ * transfer operators of type MGTwoLevelTransfer between each level.
+ *
+ * This class currently only works for tensor-product finite elements based on
+ * FE_Q and FE_DGQ elements. Systems involving multiple components of
+ * one of these element, as well as, systems with different elements or other
+ * elements are currently not implemented.
+ */
+template <typename MatrixType, typename VectorType>
+class MGTransferGlobalCoarsening : public dealii::MGTransferBase<VectorType>
+{
+public:
+  static_assert(std::is_same<typename MatrixType::value_type,
+                             typename VectorType::value_type>::value,
+                "Types do not match.");
+
+  /**
+   * Dimension.
+   */
+  static const int dim = MatrixType::dim;
+
+  /**
+   * Value type.
+   */
+  using Number = typename MatrixType::value_type;
+
+  /**
+   * Constructor taking an operator for each l level (minimum requirement is
```

Comment:
I am not sure I understand what `l` refers to. I guess it is the global number of levels. Maybe skip, same in the line below?
```suggestion
   * Constructor taking an operator for each level (minimum requirement is
```